src/dvclive/live.py

```diff
@@ -328,6 +343,24 @@ def log_artifact(self, path: StrPath):
                 logger.warning(f"Failed to dvc add {path}: {e}")
                 return
 
+            if type == "model":
+                if path not in self._models:
+                    self._models[path] = {}
+                    if name is None:
+                        name = get_unmatched_name_from_path(
+                            path, [m.get("name") for m in self._models.values()]
+                        )
```

Comment:
If I written to `dvclive/dvc.yaml` already, I assume `name`s from there won't make it here, right?

Cautious that they can break the workflow: you log a single model, but the first time you run the script you create `model.pkl`, next time `models/model.pkl`, next time `dvclive/models/model.pkl`, until you run out of names.

Reply:
> If I written to dvclive/dvc.yaml already, I assume names from there won't make it here, right?

We don't load the existing `dvclive/dvc.yaml` on instantiation and we overwrite it with the data logged on the current runtime.

This is not only because of `artifacts` but also makes sense for other sections. For example, a user might remove the code that logs parameters so keeping that section in `dvclive/dvc.yaml` would not make sense/break.

> Cautious that they can break the workflow: you log a single model, but the first time you run the script you create model.pkl, next time models/model.pkl, next time dvclive/models/model.pkl, until you run out of names.

As per the above explanation, this should not happen unless the user calls `live.log_artifact` with the same name inside the same script, at which point I don't think we should try to work around it but consider it a misuse of the API.